### PR TITLE
Add default constructor when T has default constructor.

### PR DESCRIPTION
**Pull request by Shachlan**

### CODE REVIEW

**Comment by Shachlan** [comment]
@StatusReport

File: named_type_impl.hpp [hunk 1]
```diff
@@ -28,6 +28,10 @@ class FLUENT_EBCO NamedType : public Skills<NamedType<T, Parameter, Skills...>>.
     template<typename T_ = T, typename = IsNotReference<T_>>
     explicit constexpr NamedType(T&& value) : value_(std::move(value)) {}
     
+  template <typename T_ = T,
+            typename = std::enable_if<std::is_default_constructible<T>::value, void>>
+  constexpr NamedType() noexcept(std::is_nothrow_constructible<T>::value) {}
```
Comment (StatusReport) [inline named_type_impl.hpp]: Indentation

**Comment by StatusReport** [comment]
Fix, squash, merge.

File: named_type_impl.hpp (hunk 1 above)
Comment (Shachlan) [reply named_type_impl.hpp]: done.

**Comment by Shachlan** [comment]
finished squashing, I don't have merge permissions.

**Pull request merged**